lib/ask/runtime/session.ex

```diff
@@ -413,7 +471,11 @@ defmodule Ask.Runtime.Session do
         end
 
         log_prompts(reply, channel, flow.mode, respondent)
-        respondent = runtime_channel |> Channel.ask(respondent, token, reply)
+
+        :ok =
+          ChannelBroker.ask(channel.id, channel.type, runtime_channel, respondent, token, reply)
+
+        respondent = update_stats(respondent)
         {:ok, %{session | flow: flow, respondent: respondent}, reply, current_timeout(session)}
```

Comment:
As far as we could analyze with @leandroradusky, and as far as I can remember, the only respondent update was regarding the stats, and we keep doing it. But instead of doing it inside the `ask` function, now we do it in the session to avoid crashes triggered by the respondent locks.